src/CounterFileRepository.php

```diff
@@ -51,24 +53,62 @@ public function addViewByUsername(Username $username): void
             FILE_APPEND
         );
 
-        $this->incrementViewsCount($username);
+        try {
+            $this->incrementViewsCount($username);
+        } catch (\Exception $exception) {
+            // Do not throw exception, because counter could be re-calculated
+        }
     }
 
     private function incrementViewsCount(Username $username): void
     {
         $counterFilePath = $this->getCounterFilePath($username);
-        // Need to open the file in "c" mode to avoid truncating before acquiring the lock and before reading
-        $counterFile = fopen($counterFilePath, "c");
-
-        // Acquire an exclusive lock to avoid two requests causing a write at the same time
-        flock($counterFile, LOCK_EX);
-        $incremented = $this->getViewsCountByUsername($username) + 1;
-        ftruncate($counterFile, 0);
-        fwrite($counterFile, "$incremented\n");
-        fflush($counterFile);
-        // Release the exclusive lock
-        flock($counterFile, LOCK_UN);
-        fclose($counterFile);
+
+        /**
+         * Need to open the file in "c" mode to avoid truncating
+         * before acquiring the lock and before reading.
+         */
+        $counterFile = fopen($counterFilePath, self::FILE_OPEN_MODE_CREATE);
+
+        if ($counterFile === false) {
+            throw new \RuntimeException(
+                "Cannot open file `$counterFilePath` for write",
+            );
+        }
+
+        try {
+            /**
+             * Acquire an exclusive lock to avoid two requests
+             * causing write at the same time.
+             */
+            $isLockAcquired = flock($counterFile, LOCK_EX);
+
+            if ($isLockAcquired === false) {
+                throw new \RuntimeException(
+                    "Cannot acquire lock file `$counterFilePath` for write",
+                );
+            }
+
+            $fileContent = file_get_contents($counterFilePath);
```

Comment:
Didn't reused `$this->getViewsCountByUsername($username)` because it may return `0` if there is problems with file reading, it will lead to counter reset. It will be better to do not increment counter in this case.